### PR TITLE
Request tracking middleware

### PR DESCRIPTION
**Pull request by azhavoro**
- Added middleware that assigns a UUID for each request.
- Stores the request UUID in the event payload
- Returns the UUID as the `X-Request-Id` response header

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
~~- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/events/handlers.py [hunk 1]
```diff
@@ -84,26 +84,43 @@ def _get_current_user(instance=None):
...
+    request = get_current_request()
+    if request is not None:
+        return request
+
+    if isinstance(instance, rq.job.Job):
+        return instance.meta.get("request", None)
+    else:
+        rq_job = rq.get_current_job()
+        if rq_job:
+            return rq_job.meta.get("request", None)
```
Comment (Marishka17) [inline cvat/apps/events/handlers.py:97]: Should we first check if the instance is rq job?
If I try to stage a failed file import then I will see in Grafana Dashboard
```
{"message":"Exception: ..." ...,"request":{"id":"33ea5cc9-1dc7-48c4-b29f-6c1f3fd4daeb"}}
```
request with id `33ea5cc9-1dc7-48c4-b29f-6c1f3fd4daeb` but likely to be expected `615b99d2-8435-40b8-878d-6314ed6204ec` (id of failed request)
image
image

**Review by Marishka17 — COMMENTED** [review]
If I try to stage a failed file import from the Cloud storage then the request_id will be null

```
{...,"request":{"id":null}}
```

**Comment by azhavoro** [comment]
> If I try to stage a failed file import from the Cloud storage then the request_id will be null
> 
> ```
> {...,"request":{"id":null}}
> ```

fixed

**Review by Marishka17 — APPROVED** [review]
LGTM

**Pull request merged**